master/buildbot/util/service.py

```diff
@@ -264,6 +268,7 @@ def reconfigServiceWithBuildbotConfig(self, new_config):
                                     **config_sibling._config_kwargs)
 
     def setServiceParent(self, parent):
+        # XXX(sa2ajj): can we make 'master' a simple read-only property instead?
```

Comment:
Le Sun Feb 22 2015 at 2:56:34 AM, Dustin J. Mitchell <
notifications@github.com> a écrit :

> In master/buildbot/util/service.py
> <https://github.com/buildbot/buildbot/pull/1543#discussion_r25128140>:
>
> > @@ -264,6 +268,7 @@ def reconfigServiceWithBuildbotConfig(self, new_config):
> >                                      **config_sibling._config_kwargs)
> >
> >      def setServiceParent(self, parent):
> > +        # XXX(sa2ajj): can we make 'master' a simple read-only property instead?
>
> Maybe I misunderstand the proposal -- AIUI the use of the property is only
> to prevent incorrectly setting self.master -- but we've never had that
> problem. I don't think it makes sense to try to dynamically navigate back
> to the master through parent relationships on every request -- those parent
> relationships are not fixed.
>
I think they are *per class* so that would be good enough.

> What *would* be cool is some kind of "magic" master reference that we
> could get from a mixin, which somehow knows the current master. That needs
> to be magic rather than just a global variable because we should be able to
> support multiple masters running in the same process (for example, for a
> graceful restart). In synchronous code, that's pretty straightforward - a
> thread-local variable - but I'm not sure how to do it with async code.
>
I'm not sure I understand your graceful restart usecase. What would be the
advantage over just reconfig? Having two masters on the same process will
give some other problems. They will share the same masterid, would run for
the same buildrequest?
What advantage would it be over just spawning another process?

I see another usecase for several masters in same reactor. It is for
multimaster integration tests. I'd like to start that way for my wamp test
strategy (multimaster, multislave, disconnection handling, etc).



> —
> Reply to this email directly or view it on GitHub
> <https://github.com/buildbot/buildbot/pull/1543/files#r25128140>.
>